Formatter/Pool.php

```diff
@@ -39,10 +39,19 @@ class Pool
      */
     public function __construct(LoggerInterface $logger = null, $defaultFormatter = null)
     {
+        // NEXT_MAJOR: make both parameters required
```

Comment:
So I thought about it, and I think BC might be actually achievable with some `func_num_args` magic. Let me give it a try.